tests/components/add_user_to_channel_modal.test.jsx

```diff
@@ -0,0 +1,252 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {shallow} from 'enzyme';
+
+import AddUserToChannelModal from 'components/add_user_to_channel_modal/add_user_to_channel_modal';
+
+describe('components/AddUserToChannelModal', () => {
+    const baseProps = {
+        channelMembers: {},
+        user: {id: 'someUserId'},
+        onHide: jest.fn(),
+        actions: {
+            addChannelMember: jest.fn(() => {
+                return new Promise((resolve) => {
+                    process.nextTick(() => resolve());
+                });
+            }),
+            getChannelMember: jest.fn(() => {
+                return new Promise((resolve) => {
+                    process.nextTick(() => resolve());
+                });
+            }),
+        },
+    };
+
+    it('should match snapshot', () => {
+        const wrapper = shallow(
+            <AddUserToChannelModal {...baseProps}/>
+        );
+
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    it('should match snapshot when a channel is selected', () => {
+        const wrapper = shallow(
+            <AddUserToChannelModal {...baseProps}/>
+        );
+
+        wrapper.setState({inviteError: 'some error'});
+        expect(wrapper).toMatchSnapshot();
```

Comment:
We are in a process of limiting the use of `toMatchSnapshot` as much as possible, especially when the component/elements are wrapped by the `Modal` as it generates too much snapshots which are difficult to read and validate in case we have to change in the future. Instead of that, can we turn it into a more specific test assertion? Could be check the CSS class and value of the `<label>`?  Same for other cases below.  Thanks

Reply:
Yes, I prefer a more specific test